.github/workflows/ci.yml

```diff
@@ -148,46 +150,99 @@ jobs:
         with:
           enable-cache: true
 
-      - uses: denoland/setup-deno@v2
-        with:
-          deno-version: v2.x
+      - run: |
+          if [ "${{ matrix.install }}" = "slim" ]; then
+            echo "--package pydantic-ai-slim"
+          elif [ "${{ matrix.install }}" = "standard" ]; then
+            echo ""
+          elif [ "${{ matrix.install }}" = "all-extras" ]; then
+            echo "--all-extras"
+          fi
+        id: install-command
 
       - run: mkdir .coverage
 
-      # run tests with just `pydantic-ai-slim` dependencies
-      - run: uv run --package pydantic-ai-slim coverage run -m pytest -n auto --dist=loadgroup
+      - run: uv sync --group dev
+      - run: uv run ${{ steps.install-command.outputs.install-command }} coverage run -m pytest -n auto --dist=loadgroup
         env:
-          COVERAGE_FILE: .coverage/.coverage.${{ runner.os }}-py${{ matrix.python-version }}-slim
+          COVERAGE_FILE: .coverage/.coverage.${{ matrix.python-version }}-${{ matrix.install }}
 
-      - run: uv run coverage run -m pytest -n auto --dist=loadgroup
-        env:
-          COVERAGE_FILE: .coverage/.coverage.${{ runner.os }}-py${{ matrix.python-version }}-standard
+      - name: store coverage files
+        uses: actions/upload-artifact@v4
+        if: matrix.python-version != '3.9'
+        with:
+          name: coverage-${{ matrix.python-version }}-${{ matrix.install }}
+          path: .coverage
+          include-hidden-files: true
 
-      - run: uv run --all-extras coverage run -m pytest -n auto --dist=loadgroup
-        env:
-          COVERAGE_FILE: .coverage/.coverage.${{ runner.os }}-py${{ matrix.python-version }}-all-extras
+  test-lowest-versions:
+    name: test lowest versions on ${{ matrix.python-version }}
+    runs-on: ubuntu-latest
+    timeout-minutes: 10
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.10", "3.11", "3.12", "3.13"]
+    env:
+      UV_PYTHON: ${{ matrix.python-version }}
+      CI: true
+      COVERAGE_PROCESS_START: ./pyproject.toml
+    steps:
+      - uses: actions/checkout@v4
 
-      - run: uv run --all-extras python tests/import_examples.py
+      - uses: astral-sh/setup-uv@v5
+        with:
+          enable-cache: true
+
+      - uses: denoland/setup-deno@v2
+        with:
+          deno-version: v2.x
+
+      - run: mkdir .coverage
+
+      - run: uv sync --group dev
 
-      # this must run last as it modifies the environment!
       - name: test lowest versions
-        if: matrix.python-version != '3.9'
         run: |
           unset UV_FROZEN
           uv run --all-extras --resolution lowest-direct coverage run -m pytest -n auto --dist=loadgroup
         env:
-          COVERAGE_FILE: .coverage/.coverage.${{ runner.os }}-py${{ matrix.python-version }}-lowest-versions
+          COVERAGE_FILE: .coverage/.coverage.${{matrix.python-version}}-lowest-versions
```

Comment:
Just for consistency, break this up so we get a separate step for the "uv run ... pytest"

```yaml
      - run: unset UV_FROZEN
      - run: uv run --all-extras --resolution lowest-direct coverage run -m pytest -n auto --dist=loadgroup
        env:
          COVERAGE_FILE: .coverage/.coverage.${{matrix.python-version}}-lowest-versions
```